pm_kext/col/packet_cache.c

```diff
@@ -22,6 +22,18 @@
 #include "pm_utils.h"
 #include "pm_debug.h"
 
+/**
+ * @brief Retrieves the packet index located in the array
+ *
+ * @par    packetCache = packet cache to use
+ * @par    packetID    = registered packet ID
+ * @return packet index
+ *
+ */
+static UINT32 getIndexFromPacketID(PacketCache *packetCache, UINT32 packetID) {
+    return (packetID - 1) % packetCache->maxSize; // -1 because packet id starts from 1
```

Comment:
It's a ring buffer, so we could have just started at 1 instead of 0. Should not matter. ;)